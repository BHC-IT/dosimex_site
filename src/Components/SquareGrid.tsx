import * as CSS from 'csstype'

export interface IStyles {
	containerStyle?: CSS.Properties
	squareStyle?: CSS.Properties
}

interface Iprops {
	styles?: IStyles
	nbLine?: number
	nbColumn?: number
}

const square = (squareStyleReceived: CSS.Properties, key: number) => (
	<div
		key={key}
		style={{ ...squareStyle, ...squareStyleReceived }}
	/>
)

const line = (nbColumns: number, squareStyle: CSS.Properties, key: number) => {
	const squares = []
	for (let i = 0; i < nbColumns; i++) {
		squares.push(square(squareStyle, i))
	}
	return (
		<div
			key={key}
			style={lineContainer}
		>
			{squares}
		</div>
	)
}

const lines = (nbLines: number, nbColumns: number, squareStyle: CSS.Properties) => {
	const rows = []
	for (let i = 0; i < nbLines; i++) {
		rows.push(line(nbColumns, squareStyle, i))
	}
	return <div>{rows}</div>
}

const Grid = (props: Iprops) => (
	<div style={{ ...mainContainer, ...props?.styles?.containerStyle }}>
		{lines(props?.nbLine ?? 6, props?.nbColumn ?? 4, props?.styles?.squareStyle ?? {})}
	</div>
)

export default Grid

const center = {
	display: 'flex',
	justifyContent: 'center',
	alignItems: 'center',
}

const mainContainer = {
	...center,
	// backgroundColor: 'green',
}

const lineContainer = {
	...center,
	flexDirection: 'row' as 'row',
}

const squareStyle = {
	height: '10px',
	width: '10px',
	backgroundColor: 'yellow',
	margin: '1rem',
	// opacity: "0.5",
}
