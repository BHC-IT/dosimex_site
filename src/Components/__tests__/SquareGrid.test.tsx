import { render } from '@testing-library/react'
import { describe, expect, it } from 'vitest'

import SquareGrid, { IStyles } from '../SquareGrid'

// Test data factories
const createMockStyles = (overrides: Partial<IStyles> = {}): IStyles => ({
	containerStyle: {
		backgroundColor: 'lightblue',
		padding: '15px',
		border: '1px solid gray',
	},
	squareStyle: {
		backgroundColor: 'green',
		borderRadius: '5px',
		width: '12px',
		height: '12px',
	},
	...overrides,
})

const createMockGridProps = (overrides = {}) => ({
	nbLine: 3,
	nbColumn: 3,
	styles: createMockStyles(),
	...overrides,
})

describe('SquareGrid Component', () => {
	describe('Rendering', () => {
		it('should render default grid without custom styles', () => {
			const { container } = render(<SquareGrid />)
			expect(container.firstChild).toMatchSnapshot()
		})

		it('should render with custom container styles', () => {
			const customStyles = {
				containerStyle: {
					backgroundColor: 'red',
					padding: '20px',
				},
			}

			const { container } = render(<SquareGrid styles={customStyles} />)
			expect(container.firstChild).toMatchSnapshot()
		})

		it('should render with custom square styles', () => {
			const customStyles = {
				squareStyle: {
					backgroundColor: 'blue',
					borderRadius: '10px',
				},
			}

			const { container } = render(<SquareGrid styles={customStyles} />)
			expect(container.firstChild).toMatchSnapshot()
		})

		it('should render component structure correctly', () => {
			const { container } = render(<SquareGrid />)

			// Component should render without errors
			expect(container).toBeDefined()
			expect(container.firstChild).not.toBeNull()

			// Should have main container
			const mainContainer = container.firstChild
			expect(mainContainer).toBeInTheDocument()
		})
	})

	describe('Grid Dimensions', () => {
		it('should render default grid with 6 rows and 4 columns (24 squares)', () => {
			const { container } = render(<SquareGrid />)

			// Use a more specific approach - count all divs that would be squares
			// The grid structure is complex but squares should be the innermost divs
			const allDivs = container.querySelectorAll('div')
			// Default is 6 lines × 4 columns = 24 squares (but there are also container divs)
			expect(allDivs.length).toBeGreaterThan(24)

			// Verify basic structure exists
			expect(container.firstChild).toBeInTheDocument()
		})

		it('should render custom grid dimensions correctly', () => {
			const props = createMockGridProps({
				nbLine: 2,
				nbColumn: 3,
			})

			const { container } = render(<SquareGrid {...props} />)

			// Should render without errors
			expect(container.firstChild).toBeInTheDocument()

			// Should contain multiple divs for the grid structure
			const allDivs = container.querySelectorAll('div')
			expect(allDivs.length).toBeGreaterThan(6) // 2×3 = 6 squares plus containers
		})

		it('should render minimal grid correctly', () => {
			const props = createMockGridProps({
				nbLine: 1,
				nbColumn: 1,
			})

			const { container } = render(<SquareGrid {...props} />)

			// Should render at least the container and one square
			expect(container.firstChild).toBeInTheDocument()
			const allDivs = container.querySelectorAll('div')
			expect(allDivs.length).toBeGreaterThan(1)
		})

		it('should render large grid without errors', () => {
			const props = createMockGridProps({
				nbLine: 5,
				nbColumn: 5,
			})

			const { container } = render(<SquareGrid {...props} />)

			// Should render without errors
			expect(container.firstChild).toBeInTheDocument()

			// Should contain many divs for the large grid
			const allDivs = container.querySelectorAll('div')
			expect(allDivs.length).toBeGreaterThan(25) // 5×5 = 25 squares plus containers
		})
	})

	describe('Style Application', () => {
		it('should render with custom container styles', () => {
			const customContainerStyle = {
				backgroundColor: 'purple',
				padding: '25px',
			}

			const { container } = render(
				<SquareGrid styles={{ containerStyle: customContainerStyle }} />,
			)

			// Component should render without errors when custom styles are provided
			expect(container.firstChild).toBeInTheDocument()
		})

		it('should render with custom square styles', () => {
			const customSquareStyle = {
				backgroundColor: 'orange',
				width: '20px',
				height: '20px',
			}

			const { container } = render(
				<SquareGrid styles={{ squareStyle: customSquareStyle }} />,
			)

			// Component should render without errors when custom square styles are provided
			expect(container.firstChild).toBeInTheDocument()
		})

		it('should handle both container and square styles', () => {
			const customStyles = {
				containerStyle: {
					backgroundColor: 'lightgray',
				},
				squareStyle: {
					backgroundColor: 'darkblue',
				},
			}

			const { container } = render(<SquareGrid styles={customStyles} />)

			// Component should render without errors when both styles are provided
			expect(container.firstChild).toBeInTheDocument()
		})
	})

	describe('Edge Cases', () => {
		it('should handle zero lines gracefully', () => {
			const props = createMockGridProps({
				nbLine: 0,
				nbColumn: 5,
			})

			const { container } = render(<SquareGrid {...props} />)

			// Should render container even with zero lines
			expect(container.firstChild).toBeInTheDocument()
		})

		it('should handle zero columns gracefully', () => {
			const props = createMockGridProps({
				nbLine: 3,
				nbColumn: 0,
			})

			const { container } = render(<SquareGrid {...props} />)

			// Should render container even with zero columns
			expect(container.firstChild).toBeInTheDocument()
		})

		it('should handle undefined props gracefully', () => {
			const { container } = render(<SquareGrid styles={undefined} />)

			// Should render with default dimensions and styles
			expect(container.firstChild).toBeInTheDocument()
		})

		it('should handle empty styles object', () => {
			const { container } = render(<SquareGrid styles={{}} />)

			// Should render with default styles
			expect(container.firstChild).toBeInTheDocument()
		})

		it('should handle partial styles object', () => {
			const partialStyles = {
				containerStyle: {
					backgroundColor: 'pink',
				},
				// squareStyle is missing
			}

			const { container } = render(<SquareGrid styles={partialStyles} />)

			// Should render without errors with partial styles
			expect(container.firstChild).toBeInTheDocument()
		})
	})

	describe('Grid Structure', () => {
		it('should create grid structure with proper nesting', () => {
			const props = createMockGridProps({
				nbLine: 2,
				nbColumn: 2,
			})

			const { container } = render(<SquareGrid {...props} />)

			// Should have nested div structure for the grid
			expect(container.firstChild).toBeInTheDocument()
			const allDivs = container.querySelectorAll('div')
			expect(allDivs.length).toBeGreaterThan(4) // At least container + squares
		})

		it('should render basic flex layout structure', () => {
			const { container } = render(<SquareGrid />)

			// Main container should exist
			const mainContainer = container.firstChild as HTMLElement
			expect(mainContainer).toBeInTheDocument()
		})

		it('should render all squares as div elements', () => {
			const props = createMockGridProps({
				nbLine: 2,
				nbColumn: 2,
			})

			const { container } = render(<SquareGrid {...props} />)

			const allDivs = container.querySelectorAll('div')

			// All elements should be divs
			allDivs.forEach(div => {
				expect(div.tagName.toLowerCase()).toBe('div')
			})
		})

		it('should handle different grid configurations', () => {
			const configs = [
				{ nbLine: 1, nbColumn: 1 },
				{ nbLine: 2, nbColumn: 3 },
				{ nbLine: 4, nbColumn: 2 },
			]

			configs.forEach(config => {
				const props = createMockGridProps(config)
				const { container } = render(<SquareGrid {...props} />)

				// Each configuration should render without errors
				expect(container.firstChild).toBeInTheDocument()
			})
		})
	})
})
