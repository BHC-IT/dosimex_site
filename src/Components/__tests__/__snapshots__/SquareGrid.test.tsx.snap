// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SquareGrid Component > Rendering > should render default grid without custom styles 1`] = `
<div
  style="display: flex; justify-content: center; align-items: center;"
>
  <div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
  </div>
</div>
`;

exports[`SquareGrid Component > Rendering > should render with custom container styles 1`] = `
<div
  style="display: flex; justify-content: center; align-items: center; background-color: red; padding: 20px;"
>
  <div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
  </div>
</div>
`;

exports[`SquareGrid Component > Rendering > should render with custom square styles 1`] = `
<div
  style="display: flex; justify-content: center; align-items: center;"
>
  <div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
    </div>
  </div>
</div>
`;

exports[`SquareGrid Component > should render default grid without custom styles 1`] = `
<div
  style="display: flex; justify-content: center; align-items: center;"
>
  <div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
  </div>
</div>
`;

exports[`SquareGrid Component > should render with custom container styles 1`] = `
<div
  style="display: flex; justify-content: center; align-items: center; background-color: red; padding: 20px;"
>
  <div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
      <div
        style="height: 10px; width: 10px; background-color: yellow; margin: 1rem;"
      />
    </div>
  </div>
</div>
`;

exports[`SquareGrid Component > should render with custom square styles 1`] = `
<div
  style="display: flex; justify-content: center; align-items: center;"
>
  <div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
    </div>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: row;"
    >
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
      <div
        style="height: 10px; width: 10px; background-color: blue; margin: 1rem; border-radius: 10px;"
      />
    </div>
  </div>
</div>
`;
